libnest/src/config_parser.rs

```diff
@@ -0,0 +1,146 @@
+//! Nest configuration file parsing implementation.
```

Comment:
We have a module problem here.

Keep in mind that Rust's modules hierarchy is a tree. This means that modules should encapsulate each others in a logical way.

Here we have
```
        libnest
        /       \
      /           \
config        config_parser
```

This sounds weird, these modules should be linked one in the other, or something similar. They shouldn't be side by side.

I think you should move this to `config/parser.rs`, or something like that.